storage/innobase/xtrabackup/src/fil_cur.cc

```diff
@@ -136,6 +175,15 @@ xb_fil_cur_result_t xb_fil_cur_open(
   strncpy(cursor->rel_path, xb_get_relative_path(node->space, cursor->abs_path),
           sizeof(cursor->rel_path));
 
+  /* During skip copying ibd which are not in DD */
+  if (opt_orphan_check && opt_lock_ddl && srv_backup_mode && cursor->is_ibd &&
+      !fsp_is_undo_tablespace(cursor->space_id) &&
+      !fsp_is_system_or_temp_tablespace(cursor->space_id)) {
+    if (!check_for_tablespace_in_dd(cursor->rel_path, cursor->space_id)) {
```

Comment:
space name check is not necessary. As long as the space_id exists it should be ok.

The name can be an issue with these scenarios.
1. create tablespace ts1;
2. create tablespace ts2 add datafile 'bs1.ibd';
3. create table `t1 space name`(a int);

Reply:
just check space_id exists, it should be fine I believe.

Reply:
I have tested scenario you have mentioned abouve. It works fine. 
But yeah it seems right, just check for space_id should be sufficient.

Reply:
how is it possible?
1. create uuid.ibd
2. dd has name ts2 but on disk is bs1.ibd
3. dd has 't1 space name1 but on disk t1@@02space@@02name.ibd

Reply:
fixed. now just checking for space_id.

Reply:
:+1:  :ok_hand: